GLTFSDK.Samples/Deserialize/Source/main.cpp

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <GLTFSDK/GLTF.h>
+#include <GLTFSDK/GLTFResourceReader.h>
+#include <GLTFSDK/GLBResourceReader.h>
+#include <GLTFSDK/Deserialize.h>
+
+// Replace this with <filesystem> (and use std::filesystem rather than
+// std::experimental::filesystem) if your toolchain fully supports C++17
+#include <experimental/filesystem>
+
+#include <fstream>
+#include <sstream>
+#include <iostream>
+
+#include <cassert>
+#include <cstdlib>
+
+using namespace Microsoft::glTF;
+
+namespace
+{
+    // The glTF SDK is decoupled from all file I/O by the IStreamReader (and IStreamWriter)
+    // interface(s) and the C++ stream-based I/O library. This allows the glTF SDK to be used in
+    // sandboxed environments, such as WebAssembly modules and UWP apps, where any file I/O code
+    // must be platform or use-case specific.
+    class StreamReader : public IStreamReader
+    {
+    public:
+        StreamReader(std::experimental::filesystem::path pathBase) : m_pathBase(std::move(pathBase))
+        {
+            assert(m_pathBase.has_root_path());
+        }
+
+        // Resolves the relative URIs of any external resources declared in the glTF manifest
+        std::shared_ptr<std::istream> GetInputStream(const std::string& filename) const override
+        {
+            // In order to construct a valid stream:
+            //1. The filename argument will be encoded as UTF-8 so use filesystem::u8path to
+            // correctly construct a path instance.
+            //2. Generate an absolute path by concatenating m_pathBase with the specified filename
+            // path. The filesystem::operator/ uses the platform's preferred directory separator if
+            // appropriate.
+            //3. Always open the file stream in binary mode. The glTF SDK will handle any text
+            // encoding issues for us.
+            auto streamPath = m_pathBase / std::experimental::filesystem::u8path(filename);
+            auto stream = std::make_shared<std::ifstream>(streamPath, std::ios_base::binary);
+
+            // Check if the stream has no errors and is ready for I/O operations
+            if (!stream || !(*stream))
+            {
+                throw std::runtime_error("Unable to create a valid input stream for uri: " + filename);
```

Comment:
```suggestion
                throw std::runtime_error("Unable to create a valid input stream for uri: " + filename + "\n");
```

I got this error when I tried to deserialize a gltf file from the GLTFSDK.Test\Resource\gltf folder from a powershell prompt and the lack of a newline made it hard to read.

> Error! - Unable to create a valid input stream for uri: simpleTriangle.binPS D:\glTF-SDK>

Reply:
Also, why am I getting an error message when trying to deserialize one of our samples?

Reply:
I don't think that file actually exists as the test only ever deserializes the manifest